online-fishing-store/src/app/services/sort/sort.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { SortButtonDirection } from 'src/app/enums/sort-button-direction-enum';
+import { Product } from 'src/app/interfaces/product';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SortService {
+  constructor() {}
+
+  public sortByRaiting(sortDirection: string, products: Product[]): Product[] {
+    return sortDirection === SortButtonDirection.ASC
+      ? products.sort((a, b) => {
+          return a.rating - b.rating;
+        })
+      : products.sort((a, b) => {
+          return b.rating - a.rating;
+        });
+  }
+
+  public sortByPrice(sortDirection, products: Product[]): Product[] {
+    return sortDirection === SortButtonDirection.ASC
+      ? products.sort((a, b) => {
+          return a.price - b.price;
+        })
+      : products.sort((a, b) => {
+          return b.price - a.price;
+        });
+  }
+
+  public sortingByFavorites(products: Product[]): Product[] {
```

Comment:
Why all name of methods start from "sort", but here i don't know why you have wrote "sorting"??????????